controllers/users.js

```diff
@@ -16,26 +18,54 @@ module.exports.getUserById = (req, res) => {
     });
 };
 
-module.exports.createUser = (req, res) => {
-  const { name, about, avatar } = req.body;
-
-  User.create({ name, about, avatar })
+module.exports.createUser = async (req, res) => {
+  const {
+    name, about, avatar, email, password,
+  } = req.body;
+  const userTrace = await User.findOne({ email });
+  if (userTrace) {
+    res.status(409).send({ message: 'Такая почта уже зарегистрирована' });
+    return;
+  }
+  bcrypt.hash(password, 10)
+    .then((hash) => User.create({
+      name, about, avatar, email, password: hash,
+    }))
     .then((user) => res.send({ data: user }))
```

Comment:
Нужно исправить: при создании пользователя отправляется пароль.
На прошлой итерации я предложил создать отдельный метод для модели, при вызове которого будут возвращены только публичные поля. Нужно здесь его использовать:
```suggestion
    .then((user) => res.send({ data: user.omitPrivate() }))
```